v3/integrations/nrgrpc/nrgrpc_server.go

```diff
@@ -42,6 +73,173 @@ func startTransaction(ctx context.Context, app *newrelic.Application, fullMethod
 	return txn
 }
 
+//
+// ErrorHandler is the type of a gRPC status handler function.
+// Normally the supplied set of ErrorHandler functions will suffice, but
+// a custom handler may be crafted by the user and installed as a handler
+// if needed.
+//
+type ErrorHandler func(context.Context, *newrelic.Transaction, *status.Status)
+
+//
+// Internal registry of handlers associated with various
+// status codes.
+//
+type statusHandlerMap map[codes.Code]ErrorHandler
+
+//
+// interceptorStatusHandlerRegistry is the current default set of handlers
+// used by each interceptor.
+//
+var interceptorStatusHandlerRegistry = statusHandlerMap{
+	codes.OK:                 OKInterceptorStatusHandler,
+	codes.Canceled:           InfoInterceptorStatusHandler,
+	codes.Unknown:            ErrorInterceptorStatusHandler,
+	codes.InvalidArgument:    InfoInterceptorStatusHandler,
+	codes.DeadlineExceeded:   WarningInterceptorStatusHandler,
+	codes.NotFound:           InfoInterceptorStatusHandler,
+	codes.AlreadyExists:      InfoInterceptorStatusHandler,
+	codes.PermissionDenied:   WarningInterceptorStatusHandler,
+	codes.ResourceExhausted:  WarningInterceptorStatusHandler,
+	codes.FailedPrecondition: WarningInterceptorStatusHandler,
+	codes.Aborted:            WarningInterceptorStatusHandler,
+	codes.OutOfRange:         WarningInterceptorStatusHandler,
+	codes.Unimplemented:      ErrorInterceptorStatusHandler,
+	codes.Internal:           ErrorInterceptorStatusHandler,
+	codes.Unavailable:        WarningInterceptorStatusHandler,
+	codes.DataLoss:           ErrorInterceptorStatusHandler,
+	codes.Unauthenticated:    InfoInterceptorStatusHandler,
+}
```

Comment:
Looks like we have tests for `ErrorInterceptorStatusHandler` below, using `codes.DataLoss`. Would it make sense to include tests for the other handlers, as well?

Reply:
Deferring for later, @nr-swilloughby 's putting something on the Go Engineering Board so it doesn't get forgotten.